jobs/grafana/templates/bin/influxdb-datasource

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -eu
+
+GRAFANA_DATASOURCES_API="<%= p('grafana.server.protocol', 'http') %>://<%= spec.ip %>:<%= p('grafana.server.http_port') %>/api/datasources"
+GRAFANA_CREDENTIALS="<%= p('grafana.security.admin_user') %>:<%= p('grafana.security.admin_password') %>"
+DATASOURCE_ID=""
+
+<% if p('grafana.influxdb.url') %>
+  INFLUXDB_URL="<%= p('grafana.influxdb.url') %>"
+  INFLUXDB_DATABASE="<%= p('grafana.influxdb.database') %>"
+  INFLUXDB_USERNAME="<%= p('grafana.influxdb.username') %>"
+  INFLUXDB_PASSWORD="<%= p('grafana.influxdb.password') %>"
+<% end %>
+
+if [ ! -n "${INFLUXDB_URL}" ]; then
```

Comment:
If you are using `if_p` you can put all the block inside the `if_p` - `end`, then you don't need to check for an empty var using bash,